packages/mobile/src/components/comments/CommentOverflowMenu.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useState } from 'react'
+
+import {
+  CommentSectionProvider,
+  useCurrentCommentSection,
+  useDeleteComment,
+  usePinComment,
+  useReportComment
+} from '@audius/common/context'
+import { encodeHashId, removeNullable } from '@audius/common/utils'
+import type { Comment } from '@audius/sdk'
+import { Portal } from '@gorhom/portal'
+
+import { IconButton, IconKebabHorizontal } from '@audius/harmony-native'
+
+import {
+  ActionDrawerWithoutRedux,
+  type ActionDrawerRow
+} from '../action-drawer'
+
+type CommentOverflowMenuProps = {
+  comment: Comment
+}
+
+export const CommentOverflowMenu = (props: CommentOverflowMenuProps) => {
+  const {
+    comment: { id, userId, isPinned }
+  } = props
+
+  // Need isOpen and isVisible to account for the closing animation
+  const [isOpen, setIsOpen] = useState(false)
+  const [isVisible, setIsVisible] = useState(false)
+
+  const { artistId, entityId, isEntityOwner, currentUserId } =
+    useCurrentCommentSection()
+
+  // TODO: Move to context?
+  const currentUserIdString = encodeHashId(currentUserId)
```

Comment:
We may need this in other places, kinda annoying we need to compare the string id and number id. Thoughts?